timemachine/md/minimizer.py

```diff
@@ -178,38 +181,30 @@ def minimize_host_4d(mols, host_config: HostConfig, ff, mol_coords=None) -> np.n
     x0 = combined_coords
     v0 = np.zeros_like(x0)
 
-    u_impls = bind_potentials(parameterize_system(hgt, ff, 1.0))
-    x = fire_minimize(x0, u_impls, box, 50)
-
-    for lamb in np.linspace(1.0, 0, 50):
-        u_impls = bind_potentials(parameterize_system(hgt, ff, lamb))
-        # NOTE: we don't save velocities between trajectories at different lambda windows; empirically this seems to
-        # reduce the efficiency of the optimization, with more windows being required to achieve an equivalent result
-        ctxt = custom_ops.Context(x, v0, box, intg, u_impls)
+    potentials, params = parameterize_system(hgt, ff, 1.0)
+    u_impl = summed_potential_from_potentials_and_params(potentials, params)
+    bound_impls = [u_impl]
+    x = fire_minimize(x0, bound_impls, box, 50)
+
+    # No need to reconstruct the context, just change the bound potential params. Allows
+    # for preserving the velocities between windows
+    ctxt = custom_ops.Context(x, v0, box, intg, bound_impls)
+    for lamb in np.linspace(1.0, 0, 50, endpoint=False):
+        _, params = parameterize_system(hgt, ff, lamb)
+        u_impl.set_params(flatten_params(params))
         xs, _ = ctxt.multiple_steps(50)
         x = xs[-1]
 
-    u_impls = bind_potentials(parameterize_system(hgt, ff, 0.0))
-    final_coords = fire_minimize(x, u_impls, box, 50)
-    for impl in u_impls:
+    _, params = parameterize_system(hgt, ff, 0.0)
+    u_impl.set_params(flatten_params(params))
+    final_coords = fire_minimize(x, bound_impls, box, 50)
```

Comment:
nit: should we expose lambda schedule (which happens to be length-50 and linear) and n_steps_per_lambda window (which happens to be 50)?

Reply:
Exposed number of windows and steps per window here: https://github.com/proteneer/timemachine/pull/1127/commits/a354028fb0b11276d18a8a35c2e3a1ab2fd17c22